all-in-one/README.md

```diff
@@ -0,0 +1,265 @@
+    ```
+    kubectl create secret tls my-tls-secret --key <private key filename> --cert <certificate filename>
+    ```
+## Configuration
+### 1. Configuring helm charts
+
+The helm charts for the API Manager deployment are available in the [WSO2 Helm Chart Repository](https://github.com/wso2/helm-apim). You can either use the charts from the repository or clone the repository and use the charts from the local copy.
+- The helm naming convention for APIM follows a simple pattern. The following format is used for naming the resources.
+```<RELEASE_NAME>-<CHART_NAME>-<RESOURCE_NAME>```
+
+#### 1.1 Mounting Keystore and Truststore using a Kubernetes Secret
+
+- If you are not including the keystore and truststore into the docker image, you can mount them using a Kubernetes secret. Following steps shows how to mount the keystore and truststore using a Kubernetes secret.
+- Create a Kubernetes secret with the keystore and truststore files. The secret should contain the primary keystore file, secondary keystore file, internal keystore file, and the truststore file. Note that the secret should be created in the same namespace in which you will be setting up the deployment.
+- Make sure to use the same secret name when creating the secret and when configuring the helm chart.
+- If you are using a different keystore file name and alias, make sure to update the helm chart configurations accordingly.
+In addition to the primary, internal keystores and truststore files, you can also include the keystores for HTTPS transport as well.
+- Refer the following sample command to create the secret and use it in the APIM.
+  
+  ```
+  kubectl create secret generic jks-secret --from-file=wso2carbon.jks --from-file=client-truststore.jks --from-file=wso2internal.jks -n <namespace>
+  ```
+
+#### 1.2 Adding ingress controller
+
+- If you need to use cipher tool to encrypt the passwords in the secret, first you need to encrypt the passwords using the cipher tool. The cipher tool can be found in the bin directory of the product pack. The following command can be used to encrypt the password.
+  
+  ```
+  sh cipher-tool.sh -Dconfigure
+  ```
+- Also the apictl can be used to encrpypt password as well. Reference can be found in [following](https://apim.docs.wso2.com/en/latest/install-and-setup/setup/api-controller/encrypting-secrets-with-ctl/).
+- Then the encrypted values should be filled in the the relevant fields of values.yaml.
+- Since the internal keystore password should be encrypted as well, you can use the cloud provider's secret store to store the password of the internal keystore. The following section can be used to add the cloud provider's credentials to fetch the internal keystore password. Configuration for aws can be at as below.
```

Comment:
Shall we mention that the internal keystore password is required to resolve the encrypted value in runtime and therefore we need to store the value in the cloud provider's secret manager and refer that value in the helm chart?